csharp_dotnetcore/3.Welcome-Users/WelcomeUserBot.cs

```diff
@@ -0,0 +1,146 @@
+                if (didBotWelcomeUser == false)
+                {
+                    // update user state flag to reflect bot handled first user interaction.
+                    await _welcomeUserStateAccessors.DidBotWelcomedUser.SetAsync(turnContext, true);
+
+                    await turnContext.SendActivityAsync($"you first message ever to this bot ", cancellationToken: cancellationToken);
+                }
+                else
+                {
+                    var text = turnContext.Activity.Text.ToLowerInvariant();
+                    switch (text)
+                    {
+                        case "hello":
+                        case "hi":
+                            await turnContext.SendActivityAsync($"You said {text}.", cancellationToken: cancellationToken);
+                            await turnContext.SendActivityAsync(_genericMessage, cancellationToken: cancellationToken);
+                            break;
+                        case "intro":
+                        case "help":
+                            await SendIntroCardAsync(turnContext, cancellationToken);
+                            break;
+                        default:
+                            await turnContext.SendActivityAsync(_genericMessage, cancellationToken: cancellationToken);
+                            break;
+                    }
+                }
+            }
+
+            // greet when users are added to the conversation
+            else if (turnContext.Activity.Type == ActivityTypes.ConversationUpdate)
+            {
+                if (turnContext.Activity.MembersAdded.Any())
+                {
+                    // go over all new memebers, greet anyone that was not the target (recipient) of this message
+                    // the 'bot' is the recipient for events from the channel
+                    foreach (var member in turnContext.Activity.MembersAdded)
+                    {
+                        // turnContext.Activity.MembersAdded == turnContext.Activity.Recipient.Id indicates the
+                        // bot was added to the conversation
+                        if (member.Id != turnContext.Activity.Recipient.Id)
+                        {
+                            await turnContext.SendActivityAsync($"Hi there - {member.Name}. Welcome to the \'Welcome User\' Bot. This bot will introduce you to Welcoming and greeting users.", cancellationToken: cancellationToken);
+                            await turnContext.SendActivityAsync($"You are seeing this message, because the bot recieved atleast one \'ConversationUpdate\' event, indicating you (and possibly others) joined a conversation. If you are using the emulator, pressing the \'Start Over\' trigger thise event. \'ConversationUpdate\' event depends on the channel your bot is running. You can read more information https://aka.ms/about-botframewor-welcome-user", cancellationToken: cancellationToken);
```

Comment:
spelling. 
atleast => at least
thise -> this

grammar. The "depends on the channel" isn't a sentence - please reword.

Reply:
The AKA link doesn't go anywhere. I assume these are being tracked somewhere for future creation?

Reply:
spelling. The AKA link is not spelled correctly. botframewor -> botframework